File: iosApp/PrendsEtRends/I18n/AppLanguage.swift
import Foundation

public enum AppLanguage: String, CaseIterable, Identifiable {
    case fr = "fr"
    case en = "en"
    case ha = "ha"

    public var id: String { rawValue }

    public var code: String { rawValue }

    public var displayName: String {
        switch self {
        case .fr: return "Français"
        case .en: return "English"
        case .ha: return "Hausa"
        }
    }

    public var flag: String {
        switch self {
        case .fr: return "🇫🇷"
        case .en: return "🇬🇧"
        case .ha: return "🇳🇬"
        }
    }
}

File: app/src/main/java/com/example/i18n/AppLanguage.kt
package com.example.i18n

import androidx.compose.runtime.staticCompositionLocalOf

enum class AppLanguage(val code: String, val displayName: String, val flag: String) {
    FR("fr", "Français", "🇫🇷"),
    EN("en", "English", "🇬🇧"),
    HA("ha", "Hausa", "🇳🇬")
}

val LocalAppLanguage = staticCompositionLocalOf { AppLanguage.FR }
